src/__tests__/index.js

```diff
@@ -51,4 +51,42 @@ describe('#SitemapGenerator', () => {
     expect(data.lastMod).toBe(queueItem.stateData.headers['last-modified']);
     expect(data.formattedLastMod).toBe('2023-01-05');
   });
+
+  test('::parsePage ignores pages with mismatched canonical URL when ignoreCanonicalized option is on', () => {
+    const page =
+      '<!doctype html><html class="no-js" lang="en-US"><head><link rel="canonical" href="http://not.foo.bar" /></head><body>Hello world</body></html>';
+    const data = gen.parsePage(queueItem, page, true);
+
+    expect(data.ignored).toBe(true);
+  });
+
+  test('::parsePage intakes pages with missing canonical URL regardless of ignoreCanonicalized option', () => {
+    const page =
+      '<!doctype html><html class="no-js" lang="en-US"><head></head><body>Hello world</body></html>';
+    const data = gen.parsePage(queueItem, page, true);
+
+    expect(data.url).toBe(queueItem.url);
+
+    const ignoreCanonicalizedOff = SitemapGenerator('http://foo.bar', {
+      ignoreCanonicalized: false
+    });
+    const dataOff = ignoreCanonicalizedOff.parsePage(queueItem, page, true);
+
+    expect(dataOff.url).toBe(queueItem.url);
```

Comment:
```suggestion
    expect(dataOff.url).toBe(queueItem.url);
    expect(!data.ignored).toBe(true);
```
not sure the best way to check for absence of a property

Reply:
Added `expect(dataOff).not.toHaveProperty('ignored');` (here and below)